tests/instancetype_test.go

```diff
@@ -655,6 +658,408 @@ var _ = Describe("[crit:medium][vendor:cnv-qe@redhat.com][level:component][sig-c
+				util.PanicOnError(err)
+			}
+			vm = &v1.VirtualMachine{
+				ObjectMeta: metav1.ObjectMeta{
+					GenerateName: "vm-",
+					Namespace:    util.NamespaceTestDefault,
+				},
+				Spec: v1.VirtualMachineSpec{
+					Instancetype: &v1.InstancetypeMatcher{
+						InferFromVolume: inferFromVolumeName,
+					},
+					Preference: &v1.PreferenceMatcher{
+						InferFromVolume: inferFromVolumeName,
+					},
+					Template: &v1.VirtualMachineInstanceTemplateSpec{
+						Spec: v1.VirtualMachineInstanceSpec{
+							Domain: v1.DomainSpec{},
+						},
+					},
+					Running: pointer.Bool(false),
+				},
+			}
+		})
+
+		It("should infer defaults from PersistentVolumeClaimVolumeSource", func() {
+			vm.Spec.Template.Spec.Volumes = []v1.Volume{{
+				Name: inferFromVolumeName,
+				VolumeSource: v1.VolumeSource{
+					PersistentVolumeClaim: &v1.PersistentVolumeClaimVolumeSource{
+						PersistentVolumeClaimVolumeSource: k8sv1.PersistentVolumeClaimVolumeSource{
+							ClaimName: sourcePVC.Name,
+						},
+					},
+				},
+			}}
+			checkVMhasInferredInstancetypeAndPreference()
+		})
+
+		DescribeTable("should infer defaults from DataVolume", func(dataVolume *cdiv1beta1.DataVolume, dataSource *cdiv1beta1.DataSource) {
+			// Optional DataSource referenced by a DataVolumeSourceRef
+			if dataSource != nil {
+				dataSource, err = virtClient.CdiClient().CdiV1beta1().DataSources(util.NamespaceTestDefault).Create(context.Background(), dataSource, metav1.CreateOptions{})
+				Expect(err).ToNot(HaveOccurred())
+
+				// Add the now generated DataSource name to the DataVolume
+				dataVolume.Spec.SourceRef.Name = dataSource.Name
+			}
+
+			dataVolume, err = virtClient.CdiClient().CdiV1beta1().DataVolumes(util.NamespaceTestDefault).Create(context.Background(), dataVolume, metav1.CreateOptions{})
+			Expect(err).ToNot(HaveOccurred())
+
+			vm.Spec.Template.Spec.Volumes = []v1.Volume{{
+				Name: inferFromVolumeName,
+				VolumeSource: v1.VolumeSource{
+					DataVolume: &v1.DataVolumeSource{
+						Name: dataVolume.Name,
+					},
+				},
+			}}
+			checkVMhasInferredInstancetypeAndPreference()
+		},
+			Entry("with annotations",
+				&cdiv1beta1.DataVolume{
+					ObjectMeta: metav1.ObjectMeta{
+						GenerateName: "datavolume-",
+						Namespace:    util.NamespaceTestDefault,
+						Annotations: map[string]string{
+							instancetypeapi.DefaultInstancetypeAnnotation:     instancetypeName,
+							instancetypeapi.DefaultInstancetypeKindAnnotation: instancetypeapi.SingularResourceName,
+							instancetypeapi.DefaultPreferenceAnnotation:       preferenceName,
+							instancetypeapi.DefaultPreferenceKindAnnotation:   instancetypeapi.SingularPreferenceResourceName,
+						},
+					},
+					Spec: cdiv1beta1.DataVolumeSpec{
+						Source: &cdiv1beta1.DataVolumeSource{
+							PVC: &cdiv1beta1.DataVolumeSourcePVC{
+								Namespace: util.NamespaceTestDefault,
+								Name:      pvcSourceName,
+							},
+						},
+						Storage: &cdiv1beta1.StorageSpec{
```

Comment:
It may be possible to leave storage empty, so the relevant parameters are derived from the cloned PVC.